### PR TITLE
Init the project's document

### PR DESCRIPTION
**Pull request by vil1**
This pull request is also intended as a mean of discussion prior the upcoming design sessions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+
+A purely-functional library for defining type-safe schemas for algebraic data types, providing free generators, SQL queries, JSON codecs, binary codecs, and migration from this schema definition.
+
+## Introduction & Highlights
+
+Scalaz Schema defines a generic representation of algebraic data structures and combinators that turn such schema into a generic computation over arbitrary data. In other words, Scalaz Schema provides a way to express any computation that abstracts over the structure of data, such as:
+
+* Codecs: given a serial format (binary, JSON, etc.) and the schema of a data structure (say, an ADT) we can derive a codec for that data structure and serial format.
+* Data Generators: given any schema we an derive random data generators (eg. scalacheck's `Gen`) that produce data satisfying that schema.
+* Schema/Data Migrations: since schemas are values, we can easily verify wether two versions of a schema are forward/backward compatible and provide a generic way to upgrade/downgrade data from one version of the schema to the other.
```
Comment (LGLO) [inline README.md]: wether -> whether

**Comment by LGLO** [comment]
Are we going to use macros?

**Comment by beezee** [comment]
@LGLO I would hope we can do this without them.

**Comment by beezee** [comment]
@vil1 I'm not sure the right way to phrase it, but I'd hope we can make this extensible in such a way that users can define typeclasses in terms of this rather easily (similar to how derivation works.)

Same as we list multiple "backends" for a schema (codecs, generators, sql) I'd love to see us prioritize a reasonable interface for users of the library to add their own "backends" and extend the functionality that can be derived from a schema definition.

**Comment by vil1** [comment]
@LGLO, as @beezee I think we should (and probably can) avoid using macros, or at least keep their use to the bare minimum (translating a case class' structure into a schema).

@beezee not sure I get your point, but I do feel that we should have a separate `Representation` abstraction that would allow to have the same schema define the structure of Scala ADTs, Avro's `GenericContainer` and so on. In any case, this should be discussed around the (upcoming) design document.

File: README.md (hunk 1 above)
Comment (vil1) [reply README.md]: good catch! I will add a commit to fix typos like that (but I'd prefer to wait before all those typos are detected)

File: README.md [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+* Diffing/Patching: given a schema we can generically compute the difference between two pieces of data satisfying that schema. In the same spirit, we have generic way to apply patches to apply patches to arbitrary data structures.
```
Comment (pierangeloc) [inline README.md]: typo: `to apply patches to apply patches`

Comment (vil1) [reply README.md]: "copy that" if I may say ^^

File: README.md [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+## Goal
+
+A purely-functional library for defining type-safe schemas for algebraic data types, providing free generators, SQL queries, JSON codecs, binary codecs, and migration from this schema definition.
+
+## Introduction & Highlights
+
+Scalaz Schema defines a generic representation of algebraic data structures and combinators that turn such schema into a generic computation over arbitrary data. In other words, Scalaz Schema provides a way to express any computation that abstracts over the structure of data, such as:
+
+* Codecs: given a serial format (binary, JSON, etc.) and the schema of a data structure (say, an ADT) we can derive a codec for that data structure and serial format.
+* Data Generators: given any schema we an derive random data generators (eg. scalacheck's `Gen`) that produce data satisfying that schema.
```
Comment (nicodom) [inline README.md]: `we an derive` -> `we can derive`

**Pull request merged**